internal/server/data/destinationcredential.go

```diff
@@ -33,7 +34,9 @@ func CreateDestinationCredential(tx WriteTxn, cr *models.DestinationCredential)
 		return handleError(err)
 	}
 
-	return nil
+	stmt := `SELECT pg_notify(current_schema() || '.dcredReq.' || ?, '')`
```

Comment:
Basic question, what is `.dcredReq.` here? A pattern that matches all credential long polling?

Reply:
Good question! 

It's unfortunate, but the max length for a channel name is 63 characters (https://www.postgresql.org/docs/current/sql-syntax-lexical.html#SQL-SYNTAX-IDENTIFIERS).

We need the schema name in here to prevent tests from interfering with each other (channels are not scoped to a schema like tables, and indexes), and that schema name can be pretty long as well, which limits the channel name even further.

We also have two base58 encoded identifiers, which each take up to 11 characters, so with the dot separators that's 24 characters.
 
We want to allow at least 22 characters for the schema name, so that only gives us ~15 or so for the channel name. Our tests regularly use schema names of 21 characters (maybe more I'll need to check again), and we don't know what schema names people will use for self-hosted installs.

This `dcredReq` is short for `DestinationCredentialRequest`, which is too long for use here.

We could change this to `destCredRequest` which only allows for ~22 characters for the schema name. That's probably ok, but is a bit close to the limit. `destCredReq` might be a little safer.

I can also adjust the names we use in tests to give us more space, but that also limits the schema name limit for self-hosted installs.

Reply:
I've changed these to `destCredReq` and `destCredResp`, and documented the size limit in the godoc for the `ListenChannelDescriptor` interface.